definitions/artifacts/mongo-authentication.json

```diff
@@ -20,7 +20,17 @@
       ],
       "properties": {
         "infrastructure": {
-          "$ref": "../types/k8s-application-infrastructure.json"
+          "title": "Mongo Cluster Infrastructure",
+          "description": "Mongo cluster infrastructure configuration.",
+          "type": "object",
+          "oneOf": [
+            {
+              "$ref": "../types/k8s-application-infrastructure.json"
+            },
+            {
+              "$ref": "../types/azure-resource-id.json"
```

Comment:
I'll play with this locally when I'm not on my phone but I think we need more here. 

Something like 

{
  "arn": {
    "$ref": ""
  }
}

Reply:
Good catch!  @WillBeebe is correct here and below. The way this is currently written it would turn infrastructure into a **field** that contains an ARI string instead of an **object** that contains a field named `ari`.  @mclacore I think you can just use this type instead: https://github.com/massdriver-cloud/artifact-definitions/blob/main/definitions/types/azure-infrastructure-database.json

That will set it up to be an object with one field (`ari` that fits the `azure-resource-id type`).

As a side note, @WillBeebe what do you think about renaming that type to be `azure-infrastructure-ari.json` or something like that. Instead of tying it to just a "databases" its just in infrastructure block that contains an `ari`.  We could do the same for AWS (`aws-infrastructure-arn.json`) and GCP (`gcp-infrastructure-grn.json`).

Reply:
I like that rename `azure-infrastructure-ari.json` and the idea to add ones like that for the other clouds. I can take that second task for AWS and GCP. 

W/ that change, the `oneOf` here becomes simpler too 👍

Reply:
https://github.com/massdriver-cloud/artifact-definitions/pull/84

Reply:
@mclacore you should make the same changes for Azure (either in this PR or in a follow up PR).